### PR TITLE
use requests library for making HTTP requests

### PR DESCRIPTION
**Pull request by he334**

### CODE REVIEW

File: flask_oauthlib/client.py [hunk 1]
```diff
@@ -398,14 +404,14 @@ def http_request(uri, headers=None, data=None, method=None):
...
 
         log.debug('Request %r with %r method' % (uri, method))
-        req = http.Request(uri, headers=headers, data=data)
-        req.get_method = lambda: method.upper()
         try:
-            resp = http.urlopen(req)
-            content = resp.read()
-            resp.close()
+            s = Session()
+            req = Request(method.upper(), uri, data=data, headers=headers)
```
Comment (EricSchles) [inline flask_oauthlib/client.py:412]: Do we handle method==None explicitly?

File: requirements.txt [hunk 2]
```diff
@@ -1,5 +1,6 @@
 Flask==0.11.1
 mock==2.0.0
 oauthlib==1.1.2
+requests==2.18.4
```
Comment (EricSchles) [inline requirements.txt:4]: 🎉

File: tests/test_client.py [hunk 3]
```diff
@@ -12,6 +14,8 @@
     from urllib import request as http
     http_urlopen = 'urllib.request.urlopen'
 
+http_urlopen = 'requests.Session.send'
+
```
Comment (EricSchles) [inline tests/test_client.py:18]: Why are we doing this, this way again?

File: tests/test_client.py [hunk 4]
```diff
@@ -147,7 +151,7 @@ def test_http_request(self, urlopen):
         )
 
         resp, content = OAuthRemoteApp.http_request('http://example.com')
-        assert resp.code == 200
+        assert resp.status_code == 200
         assert b'foo' in content
```
Comment (EricSchles) [inline tests/test_client.py:155]: looks good!

File: tests/test_client.py (hunk 3 above)
Comment (he334) [reply tests/test_client.py:18]: We are using http_urlopen in test_http_request in order to use @patch to mock a Response

Comment (EricSchles) [reply tests/test_client.py:18]: oh okay :face_palm:  I might fix up some of this in future PRs

**Pull request merged**